pkg/hintrunner/utils/math_utils.go

```diff
@@ -1,11 +1,33 @@
 package utils
 
 import (
+	"errors"
 	"math/big"
 
 	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
 )
 
+func EcDoubleSlope(point [2]*big.Int, alpha, p *big.Int) (*big.Int, error) {
+	// https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/python/math_utils.py#L151
+
+	pointX := point[0]
+	pointY := point[1]
+
+	if new(big.Int).Mod(pointY, p).Cmp(big.NewInt(0)) == 0 {
+		return nil, errors.New("point[1] % p == 0")
+	}
+
+	n := big.NewInt(3)
+	n.Mul(n, pointX)
+	n.Mul(n, pointX)
+	n.Add(n, alpha)
+
+	m := big.NewInt(2)
+	m.Mul(m, pointY)
+
+	return div_mod(n, m, p)
+}
```

Comment:
If I remember correctly the initial designs for a few util functions like this did have separate parameters. But we made it like this to match the python version.

Reply:
I don't agree. In my opinion doing smth because X project did it that way is not justification enough. I don't see no performance nor readability gains from this approach. Also, I am just passing by, so I won't inssist

Reply:
Ok, so after thinking about it, I don't think this approach is correct. I planned on not giving it too much though but in reality this only obfucaste the purpose of the function. It's only two parameters, points X & Y.

You need to take into account that the Cairo VM when created was meant to be very flexible and be able to work with many paremeters. This is not the case currently.

Reply:
Makes sense.